pipeline/secret.go

```diff
@@ -119,12 +122,22 @@ func (s *Secret) ParseOrg(org string) (string, string, error) {
 	// secret is invalid
 	// nolint:gomnd // accepting magic number
 	if len(parts) != 2 {
-		return "", "", fmt.Errorf("%s: %s ", ErrInvalidPath, path)
+		return "", "", fmt.Errorf("%w: %s ", ErrInvalidPath, path)
 	}
 
 	// check if the org provided matches what we expect
 	if !strings.EqualFold(parts[0], org) {
-		return "", "", fmt.Errorf("%s: %s ", ErrInvalidOrg, parts[0])
+		return "", "", fmt.Errorf("%w: %s ", ErrInvalidOrg, parts[0])
+	}
+
+	// check if path segments empty
+	if len(parts[1]) == 0 {
+		return "", "", fmt.Errorf("%w: %s ", ErrInvalidPath, path)
+	}
+
+	// secret names can't contain restricted characters.
+	if strings.ContainsAny(s.Name, constants.SecretRestrictedCharacters) {
```

Comment:
do we really want this? wouldn't this mean that you can't have a name with an `x` or a `1`? do we need to create an array for disallowed character(s) sequences instead?

Reply:
I'll need to double check. I think it was supposed to actually be `\u0000` which is the nul character but I think when I was copying and pasting I pasted the wrong value, good catch. But it should only be the 2 characters the "null" control character and =. Technically I don't think you're able to put null characters in yaml but if in the future it changes or I haven't checked Starlark either.

Reply:
And `ContainsAny` will iterate over the string and if any character in the secondary string is part of the first it'll error. It doesn't have to match exactly and if you name a secret with an = somewhere in the middle (I don't know why you would) it would caused issues with assignment and the values containing part of the name in the value.

```yaml
- name: some=password
```
->
```sh
export some=password=value
```
So the env var name would be injected eventually as `$SOME`